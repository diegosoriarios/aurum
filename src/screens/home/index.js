import React, { useEffect, useState, useContext } from 'react';
import { View, Text, TouchableOpacity, FlatList, TextInput } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import HomeListItem from './components/HomeListItem';
import UserContext from '../../store/UserContext';
import styles from './styles';

function Home() {
  const { cases, getCases } = useContext(UserContext);

  const [searchResult, setSearchResult] = useState([]);
  const [searchString, setSearchString] = useState('');
  const [searching, setSearching] = useState(false);

  useEffect(() => {
    getCases()
  }, []);

  const handleSearch = (text) => {
    setSearchString(text);

    const searchFilter = cases.filter(ca => {
      if (ca.title.toLowerCase().includes(searchString.toLowerCase())) return ca;
    });

    setSearchResult(searchFilter);
  }

  const handleCloseSearch = () => {
    setSearching(!searching);
    setSearchString('');
  }

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        {
          searching ?
          <TextInput 
            value={searchString}
            onChangeText={(text) => handleSearch(text)}
            autoFocus={true}
          /> :
          <Text style={styles.headerText}>Processos</Text>
        }
        <TouchableOpacity onPress={handleCloseSearch}>
          <Icon name={searching ? "close" : "search"} size={30} color="#11A8F3" />
        </TouchableOpacity>
      </View>
      <FlatList
          keyExtractor={item => item.id}
          data={searching? searchResult : cases}
          renderItem={({item}) => <HomeListItem item={item} />}
        />
      
    </View>
  );
}

export default Home;