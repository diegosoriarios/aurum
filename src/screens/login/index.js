import React, { useState, useContext } from 'react';
import { View, Text, TextInput } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import CustomButtom from '../../components/CustomButton';
import UserContext from '../../store/UserContext';
import styles from './styles';

function Login({ navigation }) {
  const { email, changeEmail } = useContext(UserContext);

  const [user, setUser] = useState(email);
  const [password, setPassword] = useState('');

  const handleLogin = () => {
    changeEmail(user);
    navigation.navigate("App");
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Login</Text>
      <TextInput
        style={styles.input}
        placeholder="UsuÃ¡rio"
        value={user}
        onChangeText={text => setUser(text)}
      />
      <TextInput 
        style={styles.input}
        placeholder="Senha"
        value={password}
        onChangeText={text => setPassword(text)}
      />
      <CustomButtom text="Entrar" onPress={handleLogin} />
    </View>
  );
}

export default Login;