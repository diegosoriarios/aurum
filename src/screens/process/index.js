import React, { useState, useEffect } from 'react';
import { ScrollView, View, Text, TouchableOpacity, StyleSheet, FlatList } from 'react-native';
import Modal from 'react-native-modal';
import Icon from 'react-native-vector-icons/MaterialIcons';

function Process({route, navigation}) {
  const { title, number, court, amount, customers, historicals } = route.params;
  const {name, roleName} = customers[0];
  const [modalVisible, setModalVisible] = useState(false);
  const [ordenarPor, setOrdenarPor] = useState("Ordenar por data");
  const [tipoOrdenacao, setTipoOrdernacao] = useState("Crescente");

  useEffect(() => {
    navigation.setOptions({
      title: 'Processo',
      headerRight: () => <TouchableOpacity style={{padding: 10}}><Icon name="attach-file" size={30} color="#11A8F3" /></TouchableOpacity>,
      headerLeft: () => <TouchableOpacity style={{padding:10}} onPress={() => navigation.goBack()}><Icon name="keyboard-backspace" size={30} color="#11A8F3" /></TouchableOpacity>
    })
  }, []);

  const renderHistoric = () => {
    let historic
    if (ordenarPor === "Ordenar por data") {
      if (tipoOrdenacao === "Crescente") {
        historic = historicals.sort((a, b) => new Date(a) - new Date(b)); 
      } else {

      }
    } else {

    }

    return historicals.map(item => <ListItem item={item} />);
  }

  return (
    <ScrollView>
      <View  style={styles.container}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.subtitle}>Numero</Text>
        <Text style={styles.description}>{number}</Text>
        <Text style={styles.subtitle}>Cliente</Text>
        <Text style={styles.description}>{name}</Text>
        <Text style={styles.subtitle}>Parte</Text>
        <Text style={styles.description}>{roleName}</Text>
        <Text style={styles.subtitle}>Forum</Text>
        <Text style={styles.description}>{court}</Text>
        <Text style={styles.subtitle}>Valor</Text>
        <Text style={styles.description}>{amount}</Text>
        <Text style={styles.subtitle}>Anexo</Text>
        <View style={styles.attachmentContainer}>
          <TouchableOpacity style={styles.btnContainer}>
            <Text style={styles.btnText}>NomeDoAnexo_máx20car.pdf</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.closeContainer}>
            <Text><Icon name="close" size={30} color="#000" /></Text>
          </TouchableOpacity>
        </View>  
      </View>
      <View style={styles.divider} />
      <View style={styles.container}>
        <View style={styles.historyHeader}>
          <Text style={styles.historyHeaderTitle}>HISTÓRICO</Text>
          <TouchableOpacity style={styles.historyHeaderBtn} onPress={() => setModalVisible(true)}>
            <Text>{ordenarPor}</Text>
            <Icon name={tipoOrdenacao === "Crescente" ? "arrow-drop-down" : "arrow-drop-up"} size={30} color="#000" />
            <Icon name="sort" size={30} color="#000" />
          </TouchableOpacity>
        </View>

        { historicals && renderHistoric() }
        
      </View>

      <Modal
        swipeDirection={['up', 'left', 'right', 'down']}
        isVisible={modalVisible}
        style={styles.modal}
        onBackButtonPress={() => setModalVisible(false)}
        onBackdropPress={() => setModalVisible(false)}
      >
        <View style={styles.modalContainer}>
          <RadioGroup text="Ordenar por data" selected={ordenarPor} onSelect={setOrdenarPor} />
          <RadioGroup text="Ordenar por Descriçao" selected={ordenarPor} onSelect={setOrdenarPor} />
          <View style={styles.divider} />
          <RadioGroup text="Crescente" selected={tipoOrdenacao} onSelect={setTipoOrdernacao} />
          <RadioGroup text="Decrescente" selected={tipoOrdenacao} onSelect={setTipoOrdernacao} />
        </View>
      </Modal>
    </ScrollView>
  );
}

function ListItem({item}) {
  const months = [
    'Janeiro',
    'Fevereiro',
    'Março',
    'Abril',
    'Maio',
    'Junho',
    'Julho',
    'Agosto',
    'Setembro',
    'Outubro',
    'Novembro',
    'Dezembro'
  ]

  const { description, date, id } = item;
  const itemDate = new Date(date);
  const day = itemDate.getDate();
  const month = itemDate.getMonth();
  const year = itemDate.getFullYear();

  return (
    <View style={styles.cardContainer} key={id}>
          <View style={styles.cardContainerHeader}>
            <View style={styles.dayContainer}>
              <Text style={styles.dayContainerText}>{day}</Text>
            </View>
            <View style={styles.dateContainer}>
              <Text style={styles.monthContainer}>{months[month]}</Text>
              <Text style={styles.yearContainer}>{year}</Text>
            </View>
          </View>
          <View style={styles.content}>
            <View style={styles.dividerContainer}>
            <View style={styles.verticalDivider} /></View> 
            <View style={styles.historyDescriptionBox}>
              <Text style={styles.historyDescription}>
                {description}  
              </Text>
            </View>
          </View>
        </View>
  );
}

function RadioGroup({text, selected, onSelect}) {
  let child = text == selected ? <Icon name="check" color="#11A8F3" size={30} /> : <View />
  return (
    <TouchableOpacity style={styles.radioGroup} onPress={() => onSelect(text)}>
      <View style={styles.radioSelected}>
        {child}
      </View>
      <Text style={styles.radioGroupText}>{text}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  title: {
    fontSize: 20,
    paddingBottom: 20,
    fontFamily: 'Cabin-Bold',
  },
  subtitle: {
    fontSize: 14,
    color: "grey",
    fontFamily: 'Cabin-SemiBold',
  },
  description: {
    fontSize: 18,
    color: "black",
    paddingBottom: 10,
    fontFamily: 'Cabin-SemiBold',
  },
  attachmentContainer: {
    flexDirection: "row",
  },
  btnContainer: {
    height: 50,
    borderRadius: 8,
    marginVertical: 8,
    backgroundColor: "#11A8F3",
    alignItems: "center",
    justifyContent: "center",
    padding: 10,
  },
  btnText: {
    fontWeight: "bold",
    color: "white",
  },
  closeContainer: {
    height: 50,
    borderRadius: 8,
    marginVertical: 8,
    alignItems: "center",
    justifyContent: "center",
    padding: 10,
  },
  divider: {
    height: 1,
    width: "100%",
    backgroundColor: "grey",
  },
  historyHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 10,
  },
  historyHeaderBtn: {
    flexDirection: "row",
    alignItems: "center",
  },
  historyHeaderTitle: {
    fontSize: 20,
    fontFamily: 'Cabin-SemiBold',
  },
  cardContainer: {
    flex: 1,
  },
  cardContainerHeader: {
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  dayContainer: {
    backgroundColor: "#11A8F3",
    height: 50,
    width: 50,
    borderRadius: 8,
    justifyContent: "center",
    alignItems: "center",
  },
  dayContainerText: {
    fontSize: 20,
    fontWeight: "bold",
    color: 'white',
    fontFamily: 'Cabin-SemiBold',
  },
  dateContainer: {
    paddingHorizontal: 10,
  },
  monthContainer: {
    fontSize: 16,
    fontFamily: 'Cabin-SemiBold',
  },
  yearContainer: {
    fontSize: 12,
    color: "grey",
    fontFamily: 'Cabin-SemiBold',
  },
  content: {
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  dividerContainer: {
    width: 50,
    alignItems: 'center',
  },
  verticalDivider: {
    width: 2,
    flex: 1,
    backgroundColor: "black",
  },
  historyDescriptionBox: {
    width: "80%"
  },
  historyDescription: {
    fontSize: 18,
    paddingHorizontal: 10,
    textAlign: "justify",
    fontFamily: 'Cabin-Regular',
  },
  modal: {
    justifyContent: 'flex-end',
    margin: 0,
  },
  modalContainer: {
    borderTopLeftRadius: 8,
    borderTopRightRadius: 8,
    backgroundColor: "white",
    width: "100%",
    justifyContent: "center",
    alignItems: "flex-start",
    padding: 20,
  },
  radioGroup: {
    flexDirection: "row",
    alignItems: "center",
    height: 50,
  },
  radioGroupText: {
    fontSize: 20,
    fontFamily: 'Cabin-Bold',
  },
  radioSelected: {
    width: 50,
    height: 50,
  },
});

export default Process;