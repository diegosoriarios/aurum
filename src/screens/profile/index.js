import React from 'react';
import { useContext } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import CustomButtom from '../../components/CustomButton';
import UserContext from '../../store/UserContext';

function Profile() {
  const { email } = useContext(UserContext);
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Usu√°rio</Text>
      <Text style={styles.email}>{email}</Text>
      <CustomButtom text="Sair" onPress={() => {}} />
    </View>
  );
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'center',
    padding: 20,
  },
  title: {
    fontSize: 32,
    paddingBottom: 20,
    fontFamily: 'Cabin-Bold',
  },
  email: {
    fontSize: 20,
    paddingBottom: 20,
    fontFamily: 'Cabin-SemiBold',
    marginBottom: 100,
  },
});


export default Profile;