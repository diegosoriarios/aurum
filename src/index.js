import 'react-native-gesture-handler';
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { Login } from './screens';
import { UserProvider } from './store/UserContext';
import BottomNavigation from './routes/BottomNavigation';

const Stack = createStackNavigator();

function App() {
  return (
    <UserProvider>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Login" component={Login} options={{ headerShown: false }} />
          <Stack.Screen name="App" component={BottomNavigation} options={{ headerShown: false }} />
        </Stack.Navigator>
      </NavigationContainer>
    </UserProvider>
  );
}

export default App;