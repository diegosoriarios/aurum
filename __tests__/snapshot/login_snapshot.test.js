import React from 'react';
import renderer from 'react-test-renderer';
import { Login } from '../../src/screens';

test('renders correctly', () => {
  const navigation = {}
  const tree = renderer.create(<Login navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});