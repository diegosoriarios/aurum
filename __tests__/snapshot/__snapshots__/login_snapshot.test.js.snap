// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "paddingBottom": 40,
      }
    }
  >
    Login
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="UsuÃ¡rio"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "grey",
        "borderRadius": 8,
        "borderWidth": 1,
        "marginVertical": 8,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Senha"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "grey",
        "borderRadius": 8,
        "borderWidth": 1,
        "marginVertical": 8,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#11A8F3",
        "borderRadius": 8,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 8,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Cabin-SemiBold",
          "fontWeight": "bold",
        }
      }
    >
      ENTRAR
    </Text>
  </View>
</View>
`;
